### PR TITLE
Feature/Introduced onboarding sign in

### LINKED ISSUE #3326: Introduce onboarding "Sign in" screens
Introduce
-welcome screen
-sign in screen
-recover screen
-accounts screen

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #3326

### CODE REVIEW

**Review by goranjovic — APPROVED** [review]

**Comment by janherich** [comment]
@flexsurfer  After this PR is merged, will it mean that console chat is not shown at all, meaning the one can't issue `/faucet` command ? That's big bummer for anyone developing wallet/send-command functionality, I think it's crucial that we will still have the option to easily get testnet ETH for testing.

**Comment by goranjovic** [comment]
@flexsurfer @janherich I thought it will be shown if you set Dev mode somewhere, no?

**Comment by janherich** [comment]
@flexsurfer @goranjovic I'm concerned about this:
https://github.com/status-im/status-react/pull/3470/files#diff-50b2d2f85046da6681844adb5e140a06R61

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/status-im/status-react/pull/3470">Feature/Introduced onboarding sign in by flexsurfer · Pull Request #3470 · status-im/status-react</a></strong></div><div>fixes #3326</div></blockquote>

**Comment by flexsurfer** [comment]
yes @janherich you're right, i'll comment this after PR will be tested, because i want to merge this PR, but dev option in the profile will be implemented only in next PR

File: src/status_im/chat/events.cljs [hunk 1]
```diff
@@ -153,18 +152,15 @@
...
-             :dispatch-n [[:add-contacts [console/contact]]]
-             :save-chat console/chat
-             :save-all-contacts [console/contact]}
-
-      (not current-account-id)
-      (update :dispatch-n concat [[:chat-received-message/add-when-commands-loaded console/intro-message1]]))))
+    {:db (-> db
+             (assoc :current-chat-id constants/console-chat-id)
+             (update :chats assoc constants/console-chat-id console/chat))
+     :dispatch-n [[:add-contacts [console/contact]]]
```
Comment (jeluard) [inline src/status_im/chat/events.cljs]: Why not `dispatch` ?

File: src/status_im/chat/events.cljs [hunk 2]
```diff
@@ -153,18 +152,15 @@
...
+     :save-chat console/chat
+     :save-all-contacts [console/contact]}))
```
Comment (jeluard) [inline src/status_im/chat/events.cljs]: Alignment?

File: src/status_im/chat/events/console.cljs [hunk 3]
```diff
@@ -1,7 +1,5 @@
 (ns status-im.chat.events.console
-  (:require [re-frame.core :as re-frame]
-            [status-im.utils.handlers :as handlers]
-            [status-im.constants :as const]
+  (:require [status-im.constants :as const]
```
Comment (jeluard) [inline src/status_im/chat/events/console.cljs]: `constants` ?

File: src/status_im/chat/subs.cljs [hunk 4]
```diff
@@ -55,7 +56,10 @@
   :get-active-chats
   :<- [:get-chats]
   (fn [chats]
-    (into {} (filter (comp :is-active second)) chats)))
+    (into {} (filter #(let [chat (second %)]
+                        ;;TODO console should be shown in dev mode
+                        (and (:is-active chat) (not= const/console-chat-id (:chat-id chat))))
```
Comment (jeluard) [inline src/status_im/chat/subs.cljs]: Extract function for better lisibility?

File: src/status_im/ui/components/colors.cljs [hunk 5]
```diff
@@ -7,11 +8,19 @@
...
 (def gray-light "#e8ebec") ;; Used as divider color
 (def gray-lighter "#eef2f5")  ;; Used as a background or shadow
-(def blue "#4360df") ;; Used as main wallet color
-(def blue-transparent "rgba(67, 96, 223, 0.10)")  ;;used as shadow for blue elements
+(def blue "#4360df") ;; Used as main wallet color, and ios home add button
 (def red "#ff2d55") ;; Used to highlight errors or "dangerous" actions
+(def red-light "#ffe5ea") ;; error tooltip
 (def text-light-gray "#212121") ;; Used for labels (home items)
 
+(defn alpha [hex opacity]
```
Comment (jeluard) [inline src/status_im/ui/components/colors.cljs:19]: Nice!

File: src/status_im/ui/components/common/common.cljs [hunk 6]
```diff
@@ -7,7 +7,9 @@
             [status-im.ui.components.icons.vector-icons :as vector-icons]
             [status-im.ui.components.react :as react]
             [status-im.utils.ethereum.core :as ethereum]
-            [status-im.utils.platform :as platform]))
+            [status-im.utils.platform :as platform]
+            [status-im.ui.components.icons.vector-icons :as icons]
```
Comment (jeluard) [inline src/status_im/ui/components/common/common.cljs:11]: `vector-icons` ?

File: src/status_im/ui/components/common/common.cljs [hunk 7]
```diff
@@ -63,3 +65,25 @@
...
          (i18n/label :t/testnet-text {:testnet (get-in ethereum/chains [(ethereum/chain-id->chain-keyword network-id) :name] "Unknown")})
          (i18n/label :t/mainnet-text))]]]))
+
+(defn logo
+  ([] (logo nil))
+  ([{:keys [size icon-size shadow?] :or {shadow? true}}]
+   [react/view {:style (styles/logo-container size shadow?)}
+    [icons/icon :icons/logo (styles/logo icon-size)]]))
+
+(defn bottom-button [{:keys [label disabled? on-press forward?]}]
```
Comment (jeluard) [inline src/status_im/ui/components/common/common.cljs:75]: There is `bottom-buttons` and `button` ns already. Are those different?

File: src/status_im/ui/components/common/styles.cljs [hunk 8]
```diff
@@ -109,3 +100,51 @@
    :color         colors/blue
    :ios           {:font-size 15}
    :android       {:font-size 14}})
+
+(defstyle logo-shaddow
+  {:ios     {:shadowColor   colors/black
```
Comment (jeluard) [inline src/status_im/ui/components/common/styles.cljs:105]: No camel case?

File: src/status_im/ui/components/text_input/view.cljs [hunk 9]
```diff
@@ -0,0 +1,20 @@
+(ns status-im.ui.components.text-input.view
+  (:require [status-im.ui.components.react :as react]
+            [status-im.ui.components.text-input.styles :as styles]
+            [status-im.ui.components.colors :as colors]
+            [status-im.ui.components.tooltip.views :as tooltip]))
+
+(defn text-input-with-label [{:keys [label error style height] :as props}]
```
Comment (jeluard) [inline src/status_im/ui/components/text_input/view.cljs:7]: Do we still use this?

File: src/status_im/ui/components/toolbar/view.cljs [hunk 10]
```diff
@@ -110,21 +108,17 @@
 
 (defn toolbar
   ([props nav-item content-item] (toolbar props nav-item content-item [actions [{:image :blank}]]))
-  ([{:keys [background-color style flat? title-centered?]}
```
Comment (jeluard) [inline src/status_im/ui/components/toolbar/view.cljs:113]: No more centering?

File: src/status_im/ui/components/tooltip/styles.cljs [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+   :left        0
+   :right       0
+   :bottom      bottom-value
+   :opacity     opacity-value})
+
+(defn tooltip-text-container [color]
+  {:padding-horizontal 16
+   :padding-vertical   9
+   :background-color   color
+   :border-radius      8})
```
Comment (jeluard) [inline src/status_im/ui/components/tooltip/styles.cljs:23]: It's defined in `styles`

File: src/status_im/ui/screens/accounts/create/styles.cljs [hunk 12]
```diff
@@ -0,0 +1,58 @@
+(ns status-im.ui.screens.accounts.create.styles
+  (:require [status-im.ui.components.colors :as colors]))
+
+(def create-account-view
+  {:flex               1
+   :background-color colors/white})
```
Comment (jeluard) [inline src/status_im/ui/screens/accounts/create/styles.cljs:6]: Alignment

File: src/status_im/ui/screens/accounts/create/views.cljs [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+    :on-change-text    #(re-frame/dispatch [:set-in [:accounts/create (get-in steps [step :input-key])] %])
+    :secure-text-entry (boolean (#{:enter-password :confirm-password} step))
+    :error             error}])
+
+(defview create-account []
+  (letsubs [step [:get-in [:accounts/create :step]]
+            next-enabled? [:get-account-creation-next-enabled?]
+            error [:get-in [:accounts/create :error]]
+            password [:get-in [:accounts/create :password]]
+            password-confirm [:get-in [:accounts/create :password-confirm]]]
```
Comment (jeluard) [inline src/status_im/ui/screens/accounts/create/views.cljs:56]: Can't be simplified? Like a single sub?

File: src/status_im/ui/screens/accounts/create/views.cljs [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+         [components.common/logo styles/account-creating-logo]]
+        [react/view {:style styles/account-creating-indicatior}
+         [components/activity-indicator {:animating true}]
+         [react/text {:style styles/account-creating-text}
+          (i18n/label :t/creating-your-account)]]])
+     (when (#{:enter-password :confirm-password :enter-name} step)
+       [react/view components.styles/flex
+        [toolbar/toolbar {:flat? true} (when (#{:enter-password :confirm-password} step)
+                                         (toolbar/nav-button (actions/back #(step-back step)))) nil]
+        [react/view {:style styles/logo-container}
```
Comment (jeluard) [inline src/status_im/ui/screens/accounts/create/views.cljs:71]: Would read better with sub-components

File: src/status_im/ui/screens/accounts/events.cljs [hunk 15]
```diff
@@ -90,6 +81,12 @@
                                          :private updates-private-key}}}}))))
 ;;;; Handlers
 
+(handlers/register-handler-fx
+  :create-account
+  (fn [{{:accounts/keys [create] :as db} :db}]
+    {:db (update db :accounts/create assoc :step :account-creating :error nil)
```
Comment (jeluard) [inline src/status_im/ui/screens/accounts/events.cljs:87]: Alignment

File: src/status_im/ui/screens/accounts/events.cljs [hunk 16]
```diff
@@ -204,3 +193,20 @@
         ;; TODO(janherich): this is very strange and misleading, need to figure out why it'd necessary to update
         ;; account with network update when last update was more then week ago
         (account-update {:db db} nil)))))
+
+(handlers/register-handler-fx
+  :account-set-name
+  (fn [{{:accounts/keys [create] :as db} :db} _]
+    (-> {:db (assoc-in db [:accounts/create :show-welcome?] true)
```
Comment (jeluard) [inline src/status_im/ui/screens/accounts/events.cljs:200]: Alignment

File: src/status_im/ui/screens/accounts/login/views.cljs [hunk 17]
```diff
@@ -37,31 +39,52 @@
     :else
     :t/unknown-status-go-error))
 
+(defn account-login-badge [photo-path name]
+  [react/view styles/login-badge
+   [react/image {:source {:uri (if (string/blank? photo-path) :avatar photo-path)}
+                 :style  styles/login-badge-image}]
+   [react/view
+    [react/text {:style         styles/login-badge-name
+                 :numberOfLines 1}
```
Comment (jeluard) [inline src/status_im/ui/screens/accounts/login/views.cljs:48]: No kebab case?

File: src/status_im/ui/screens/accounts/login/views.cljs [hunk 18]
```diff
@@ -37,31 +39,52 @@
...
+       [account-login-badge photo-path name]
+       [react/view styles/password-container
+        [text-input/text-input-with-label
+         {:label             (i18n/label :t/password)
+          :placeholder       (i18n/label :t/password)
+          :ref               #(reset! password-text-input %)
+          :on-submit-editing #(login-account @password-text-input address password)
+          :on-change-text    #(do
+                                (re-frame/dispatch [:set-in [:accounts/login :password] %])
+                                (re-frame/dispatch [:set-in [:accounts/login :error] ""]))
```
Comment (jeluard) [inline src/status_im/ui/screens/accounts/login/views.cljs]: Can't be a dedicated handler for both info?

File: src/status_im/ui/screens/accounts/styles.cljs [hunk 19]
```diff
@@ -21,47 +27,39 @@
...
-  {:margin-left 16
-   :flex-shrink 1})
+  {:margin-left  16
+   :margin-right 21
+   :flex-shrink  1})
+
+(def account-badge-text
+  {:font-size      17
+   :letter-spacing -0.2
+   :color          common/color-black})
```
Comment (jeluard) [inline src/status_im/ui/screens/accounts/styles.cljs:49]: `colors/black`

File: src/status_im/ui/screens/intro/styles.cljs [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+   :justify-content :center})
+
+(def intro-logo
+  {:size      111
+   :icon-size 46})
+
+(defstyle intro-text
+  {:text-align  :center
+   :color       colors/black
+   :ios         {:line-height 28
```
Comment (jeluard) [inline src/status_im/ui/screens/intro/styles.cljs]: Alignment

File: src/status_im/ui/screens/intro/styles.cljs [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+                 :font-size      22
+                 :font-weight    :bold
+                 :letter-spacing -0.3}
+   :android     {:font-size 24
```
Comment (jeluard) [inline src/status_im/ui/screens/intro/styles.cljs]: Alignment

File: src/status_im/ui/screens/usage_data/styles.cljs [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+
+(def usage-data-image
+  {:width      138
+   :height     208
+   :margin-top 10})
+
+(defstyle help-improve-text
+  {:text-align  :center
+   :color       colors/black
+   :ios         {:line-height 28
```
Comment (jeluard) [inline src/status_im/ui/screens/usage_data/styles.cljs]: Alignment

File: src/status_im/ui/screens/usage_data/styles.cljs [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+                 :font-size      22
+                 :font-weight    :bold
+                 :letter-spacing -0.3}
+   :android     {:font-size 24
```
Comment (jeluard) [inline src/status_im/ui/screens/usage_data/styles.cljs]: Alignment

File: src/status_im/ui/screens/wallet/transactions/views.cljs [hunk 24]
```diff
@@ -187,7 +187,7 @@
 (defview transactions []
   (letsubs [current-tab                 [:get :view-id]
             filter-data                 [:wallet.transactions/filters]]
-    [react/view {:style components.styles/flex}
+    [react/view {:style {:flex 1 :background-color :white}}
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/transactions/views.cljs]: Extract as style?

**Review by jeluard — CHANGES_REQUESTED** [review]
Impressive work ! 

Not a big fan of the various `:set-in`
.
Also feels like some of the styles could be merged and have some more common parts?

**Comment by yevh-berdnyk** [comment]
Android: https://i.diawi.com/aFtrrf
iOS: https://i.diawi.com/qY18Ca

Keyboard expanded after running the app on iPhone covering OK button on the warning alert
img_ba5cddfed2b3-1

Recovering account: **Passphrase** instead of **Enter your seed phrase**, typo **wrords**
screen shot 2018-02-28 at 9 03 56 pm

**Password confirmation doesn’t match password** instead of **Passwords don't match** when creating account
screen shot 2018-02-28 at 9 08 17 pm

No **learn more** on Help improve Status view
screen shot 2018-02-28 at 9 11 07 pm

**Comment by flexsurfer** [comment]
@jeluard  thanks, done!

**Comment by flexsurfer** [comment]
@yevh-berdnyk 
alert - fixed
typo - fixed
learn more - will be implemented later
all texts and labels will be revisited later
thanks

**Review by jeluard — APPROVED** [review]

**Review by janherich — APPROVED** [review]

**Pull request merged**